File: C#/1130. Minimum Cost Tree From Leaf Values.cs
public class Solution {
    public int MctFromLeafValues(int[] arr) {
        int res = 0;
        Stack<int> stk = new Stack<int>();
        stk.Push(Int32.MaxValue);
        foreach(int i in arr){
            while(stk.Peek()<=i){
                int m = stk.Pop();
                res += m*Math.Min(stk.Peek(),i);
            }
            stk.Push(i);
        }
        while(stk.Count>2){
            res += stk.Pop()*stk.Peek();
        }
        return res;
    }
}

File: Java/1130. Minimum Cost Tree From Leaf Values.java
class Solution {
    public int mctFromLeafValues(int[] arr) {
        int res = 0;
        Stack<Integer> stk = new Stack<Integer>();
        stk.push(Integer.MAX_VALUE);
        for(int i: arr){
            while(stk.peek()<=i){
                int m = stk.pop();
                res += m*Math.min(stk.peek(),i);
            }
            stk.push(i);
        }
        while(stk.size()>2){
            res += stk.pop()*stk.peek();
        }
        return res;
    }
}
